### PR TITLE
[pigeon] Create the skeleton for Windows unit tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
Creates a Windows plugin, with unit tests configured using the model that has been used for flutter/plugins Windows unit tests. This was created by:
- Running `flutter create --platforms=windows -t plugin`.
- Replacing the method channel with a placeholder.
- Adding the CMake logic to build unit tests, along with a placeholder unit test.

This is intended to be used as the foundation for testing Pigeon-generated Windows platform channels, similar to the existing native Pigeon tests for iOS and Android.

Part of https://github.com/flutter/flutter/issues/91061

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
The next step would be a PR that teaches `run_test.sh` to handle Windows in a minimal way, and enables that in the CI scripts. Since you're the one who knows that script, you're probably best equipped to do that step. (I can help with the CI part, but it should be trivial once the script works.)

The way you'd build and run tests is:
- `flutter build windows --debug` in the example directory, then
- `build\windows\plugins\windows_unit_tests\Debug\windows_unit_tests_test.exe`

**Comment by stuartmorgan** [comment]
> It looks like this pull request may not have tests.

I forgot that for consistency with the existing `.dart` patterns, the bot only considers C/C++ files to be tests if they end in `_test`, not `_tests`. I've renamed the test file to follow that expectation.

File: packages/pigeon/platform_tests/windows_unit_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# windows_unit_tests
+
+A new flutter plugin project.
```
Comment (gaaclarke) [inline packages/pigeon/platform_tests/windows_unit_tests/README.md:3]: Probably should update this description.  A note with how to execute the tests would be helpful too.

File: packages/pigeon/platform_tests/windows_unit_tests/example/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# windows_unit_tests_example
+
+Demonstrates how to use the windows_unit_tests plugin.
```
Comment (gaaclarke) [inline packages/pigeon/platform_tests/windows_unit_tests/example/README.md:3]: same here

File: packages/pigeon/platform_tests/windows_unit_tests/windows/test/pigeon_test.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (gaaclarke) [inline packages/pigeon/platform_tests/windows_unit_tests/windows/test/pigeon_test.cpp:1]: We should probably point to this file in the readme.

**Review by gaaclarke — APPROVED** [review]
LGTM, just need some updated md files.  Some of this stuff seems like it comes from the template, should we not check it in and run `flutter create`?

**Pull request merged**